File: Tugas Modul 4 (ROS)/bev_workspace/src/bev_package/src/simple_publisher_node.cpp
#include "ros/ros.h"
#include "bev_package/Pesan.h"

int main(int argcm, char**argv) {
    ros::init(argcm, argv,"simple_publisher_node");

    ros::NodeHandle nh;

    ros::Publisher data_publisher = nh.advertise<bev_package::Pesan>("topik_utama",1000);

    ros::Rate loop_rate(1);

    while(ros::ok()) {
        bev_package::Pesan dataVariabel;

        dataVariabel.usia = 18;
        dataVariabel.tinggi = 163.0;
        dataVariabel.nama = "Bevinda Vivian";
        dataVariabel.kehadiran = true;

        ROS_INFO("\nUsia: %ld\nTinggi: %lf\nNama: %s\nKehadiran: %s",dataVariabel.usia, dataVariabel.tinggi, dataVariabel.nama.c_str(), dataVariabel.kehadiran ? "true" : "false");

        data_publisher.publish(dataVariabel);

        ros::spinOnce();
        loop_rate.sleep();
    }
    return 0;
}

File: Tugas Modul 4 (ROS)/bev_workspace/src/bev_package/script/simple_publisher_node.py
import rospy
from bev_package.msg import Pesan

def callbackHandler(msg):
    rospy.loginfo("\nUsia: %d\nTinggi: %lf\nNama: %s\nKehadiran: %s",msg.usia, msg.tinggi, msg.nama, "true" if msg.kehadiran else "false")

if __name__ == '__main__':
    rospy.init_node('simple_publisher_node', anonymous=True)
    data_publisher = rospy.Publisher("topik_utama", Pesan, queue_size=1000)
    rate = rospy.Rate(1)
    while not rospy.is_shutdown():
        dataVariabel = Pesan()
        dataVariabel.usia = 18
        dataVariabel.tinggi = 163.0
        dataVariabel.nama = "Bevinda Vivian"
        dataVariabel.kehadiran = True
        rospy.loginfo("\nUsia: %d\nTinggi: %lf\nNama: %s\nKehadiran: %s",dataVariabel.usia, dataVariabel.tinggi, dataVariabel.nama, "true" if dataVariabel.kehadiran else "false")
        data_publisher.publish(dataVariabel)
        rate.sleep()
